app/src/main/java/org/systers/mentorship/view/fragments/EditProfileFragment.kt

```diff
@@ -97,4 +105,20 @@ class EditProfileFragment: DialogFragment() {
         profileViewModel.successfulUpdate.removeObservers(activity!!)
         profileViewModel.successfulUpdate.value = null
     }
+
+    /**
+     * A function to perform input validation
+     * on the data entered by the user.
+     * (Add more parameters as needed
+     * in the future)
+     * @param name The user's name
+     * @return Return an array of errors.
+     * If there are no errors, the array will
+     * be empty.
+     */
+    private fun validateProfileInput(name: String?): Array<EditProfileFragmentErrorStates> {
+        var errors = arrayOf<EditProfileFragmentErrorStates>()
```

Comment:
I like this modular implementation :)

Reply:
Thank you! I wanted this feature to extensible, so I opted for this approach.